rootio/streamers.go

```diff
@@ -61,6 +61,25 @@ func (tsi *tstreamerInfo) Elements() []StreamerElement {
 	return tsi.elems
 }
 
+func (tsi *tstreamerInfo) MarshalROOT(w *WBuffer) (int, error) {
+	if w.err != nil {
+		return 0, w.err
+	}
+	pos := w.Pos()
+
+	w.WriteVersion(tsi.rvers)
+	tsi.named.MarshalROOT(w)
+	w.WriteU32(tsi.chksum)
+	w.WriteI32(tsi.clsver)
+
+	tsi.objarr.arr = make([]Object, tsi.objarr.Len())
```

Comment:
there are 2 issues with this code:
- it just write `n` `nil` pointers where `n := tsi.objarr.Len()`. in the case at hand, as `tsi.objarr.arr` has been `nil`-ed at line 108 in `tstreamerInfo.UnmarshalROOT`, it will write 0 elements. what you need is the length of the `tsi.elems` slice. _ie_ the slice that contains the informations that have been passed from `tsi.objarr.arr`, in `tstreamerInfo.UnmarshalROOT`.
- you're writing `n` elements, instead of writing a container holding (among other things) `n` elements. from the point of view of ROOT, that's really not the same thing.